pkg/controller/node/range_allocator.go

```diff
@@ -239,9 +242,20 @@ func (r *rangeAllocator) updateCIDRAllocation(data nodeAndCIDR) error {
 			}
 			return nil
 		}
+		oldData, err := json.Marshal(node)
+		if err != nil {
+			return fmt.Errorf("Failed to json.Marshal(%#v) for node %q: %v", node, data.nodeName, err)
+		}
+
 		node.Spec.PodCIDR = data.cidr.String()
-		if _, err := r.client.Core().Nodes().Update(node); err != nil {
-			glog.Errorf("Failed while updating Node.Spec.PodCIDR (%d retries left): %v", podCIDRUpdateRetry-rep-1, err)
+
+		newData, err := json.Marshal(node)
+		if err != nil {
+			return fmt.Errorf("Failed to json.Marshal(%#v) for updated node %q: %v", node, data.nodeName, err)
+		}
+
+		if err := patchNode(r.client, oldData, newData, data.nodeName); err != nil {
+			glog.Errorf("Failed while patching Node.Spec.PodCIDR for node %v (%d retries): %v", node, rep+1, err)
```

Comment:
Failed to patch Node.Spec.PodCIDR for node %v (% retries): %v

("Failed while patching" is somewhat awkward description)